fix(curriculum,a11y): improve accessibility of Drum Machine demo

Om7035 opened the pull request:
This PR fixes accessibility issues in the Drum Machine project demo as described in issue #63033:

- Replaced pad `<div>` elements with semantic `<button>` controls for better keyboard & screen-reader access
- Changed power toggle and bank toggle to `<input type="checkbox">` with proper labels
- Added `<label>` for the volume slider input
- Improved color contrast of pads from `#444` to `#666` to meet WCAG AA standards against `#3b3b5f` container
- Added test to ensure pads are displayed as buttons
- Added focus states and ARIA labels for all interactive controls

### Screenshot



<img width="1887" height="906" alt="drum-machine-a11y-demo png" src="https://github.com/user-attachments/assets/ba9fc03b-2453-4cbc-8346-cd41991d58f7" />


*The updated demo showing button elements for pads with improved color contrast (`#666` vs `#3b3b5f`)*

---

Checklist:
- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitHub Codespaces.

Closes #63033

ilenia-magoni commented:
Thank you for the contribution!

Please help add back the following template to the PR description, and follow the checklist. Until then, we are not able to review this PR.

```md
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [ ] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [ ] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [ ] My pull request targets the `main` branch of freeCodeCamp.
- [ ] I have tested these changes either locally on my machine, or GitHub Codespaces.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #XXXXX

<!-- Feel free to add any additional description of changes below this line -->
```

Om7035 commented:
> Thank you for the contribution!
> 
> Please help add back the following template to the PR description, and follow the checklist. Until then, we are not able to review this PR.
> 
> ```
> Checklist:
> 
> <!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->
> 
> - [ ] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
> - [ ] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
> - [ ] My pull request targets the `main` branch of freeCodeCamp.
> - [ ] I have tested these changes either locally on my machine, or GitHub Codespaces.
> 
> <!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->
> 
> Closes #XXXXX
> 
> <!-- Feel free to add any additional description of changes below this line -->
> ```

@ilenia-magoni  done!!!

Om7035 committed:
fix(a11y): improve accessibility of Drum Machine demo (#63033)

- Replace pad div elements with semantic button controls for better keyboard & screen-reader access
- Change power toggle and bank toggle to input type="checkbox" with proper labels
- Add label for the volume slider input
- Improve color contrast of pads from #444 to #666 to meet WCAG AA standards against #3b3b5f container
- Add test to ensure pads are displayed as buttons
- Add focus states and ARIA labels for all interactive controls

huyenltnguyen reviewed (commented):
Please update the PR description to:
- Wrap the color codes in backticks so GitHub doesn’t link to unrelated issues
- Add a screenshot of the updated demo

Om7035 committed:
fix: remove redundant aria-label from volume slider

The volume slider already has a visible label element, so the aria-label is redundant. Keeping aria-label only for checkboxes that don't have visible text labels.

Om7035 commented:
> that was a misclick, sorry

I am new to open source thats why i was taking AI's help :)

huyenltnguyen committed:
Merge branch 'main' into fix/a11y-drum-machine-63033

huyenltnguyen committed:
fix: test text

huyenltnguyen committed:
fix: active and inactive styles of toggles

huyenltnguyen committed:
fix: remove aria-label from pad buttons

huyenltnguyen committed:
fix: toggle label and styles

huyenltnguyen committed:
fix: change :focus to :focus-visible

huyenltnguyen committed:
fix: remove redundant code
